### PR TITLE
CAKE-5575 | Use new proxy host

### PR DESCRIPTION
**Pull request by JoshuaRogan**
Route all affiliate links through the affiliate service. 

@Wikia/cake

### CODE REVIEW

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+	var tracking = {
+		algo: '',
+		method: '',
+		version: '',
+		recommendation_level: '',
+	};
+
+	if (unit.tracking && unit.tracking.forEach && typeof unit.tracking.forEach === 'function') {
+		unit.tracking.forEach((kv) => {
+			tracking[`${kv.key}`] = kv.val;
```
Comment (vforge) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: Please don't use backticks in ES5; also those are not needed here

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+	return tracking;
+}
+
+function linkToProxyLink(link, unit, wikiId, articleId, host) {
+	var tracking = trackingPairsToObject(unit);
+
+	var category = unit.category;
+
+	// wikiId/articleId/category/algorithm/method/version
+	var path = `${wikiId}/${articleId}/${category}/${tracking.algo}/${tracking.method}/${tracking.version}`;
```
Comment (vforge) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: plz no backticks - use `[].join('/')` instead

Comment (JoshuaRogan) [reply extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: Damn MW

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js (hunk 1 above)
Comment (JoshuaRogan) [reply extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: https://github.com/Wikia/app/pull/17805/commits/44f4b75fb1bedeb2fa531bb72e1fea85efee93ec

**Review by vforge — APPROVED** [review]

**Comment by cschucode** [comment]
I'm not sure I understand the `thanks obama` reference in your commit. 🤔

**Pull request merged**